cachi2/core/package_managers/npm.py

```diff
@@ -173,6 +185,158 @@ def get_sbom_components(self) -> list[dict[str, str]]:
         packages = self._dependencies if self.lockfile_version == 1 else self._packages
         return [{"name": package.name, "version": package.version} for package in packages]
 
+    def get_dependencies_to_download(self) -> Dict[str, Dict[str, Optional[str]]]:
+        """Return a Dict of URL dependencies to download."""
+        packages = self._dependencies if self.lockfile_version == 1 else self._packages
+        return {
+            package.name: {
+                "version": package.version,
+                "url": package.resolved_url,
+                "integrity": package.integrity,
+            }
+            for package in packages
+            if "file:" not in package.resolved_url
+        }
+
+
+def extract_git_info_npm(vcs_url: str) -> Dict[str, str]:
```

Comment:
I think this function can be private. 

(same for `clone_repo_pack_archive` and `get_dependencies`)

Reply:
ack, I changed them.